com.unity.mobile.notifications/Editor/AndroidNotificationPostProcessor.cs

```diff
@@ -1,303 +1,177 @@
-        {
-            var applicationNode = xmlDoc.SelectSingleNode("manifest/application");
-            if (applicationNode == null)
-                throw new ArgumentException(string.Format("Missing 'application' node in '{0}'.", manifestPath));
-
-            var nodes = xmlDoc.SelectNodes("manifest/application/meta-data");
-            if (nodes != null)
-            {
-                // Check if there is a 'meta-data' with the same name.
-                foreach (XmlNode node in nodes)
-                {
-                    var element = node as XmlElement;
-                    if (element == null)
-                        continue;
-
-                    var elementName = element.GetAttribute("name", kAndroidNamespaceURI);
-                    if (elementName == name)
-                    {
-                        element.SetAttribute("value", kAndroidNamespaceURI, value);
-                        return;
-                    }
-                }
-            }
-
-            XmlElement metaDataNode = xmlDoc.CreateElement("meta-data");
-            metaDataNode.SetAttribute("name", kAndroidNamespaceURI, name);
-            metaDataNode.SetAttribute("value", kAndroidNamespaceURI, value);
-
-            applicationNode.AppendChild(metaDataNode);
-        }
-
-        private static void InjectProguard(string projectPath)
-        {
-            var proguardFile = $"{projectPath}/proguard-unity.txt";
-            if (!File.Exists(proguardFile))
-            {
-                UnityEngine.Debug.LogWarning($"Proguard file {proguardFile} not found, mobile notifications package may not function");
-                return;
-            }
-
-            var lines = File.ReadAllLines(proguardFile);
-            if (InjectProguard(ref lines))
-                File.WriteAllLines(proguardFile, lines);
-        }
-
-        internal static bool InjectProguard(ref string[] lines)
-        {
-            bool manager = InjectProguard(ref lines,
-                "com.unity.androidnotifications.UnityNotificationManager",
-                "-keep class com.unity.androidnotifications.UnityNotificationManager { public *; }");
-            bool callback = InjectProguard(ref lines,
-                "com.unity.androidnotifications.NotificationCallback",
-                "-keep class com.unity.androidnotifications.NotificationCallback { *; }");
-
-            return manager || callback;
-        }
-
-        static bool InjectProguard(ref string[] lines, string search, string inject)
-        {
-            foreach (var s in lines)
-                if (s.Contains(search))
-                    return false;
-
-            lines = lines.Concat(new[] { inject }).ToArray();
-            return true;
-        }
-    }
+    public class AndroidNotificationPostProcessor : AndroidProjectFilesModifier
+    {
+        private static readonly string NotificationProguardPath = "unityLibrary/proguard-unity-notifications.txt";

```

Comment:
Not we're creating new proguard for notifications, since we don't allow updating existing proguard-unity, since it's not incremental friendly way